research/conditional/utils/conditional_trainer.py

```diff
@@ -185,16 +183,15 @@ def _train_step(
             self._log_auxiliary_losses(aux_info["losses"], step)
         self._save_weights(step)
 
-    def calculate_loss_and_maybe_optimize(
-        self, processed_batch: LLMBatch, should_optimize: bool
-    ):
+    def calculate_loss_and_maybe_optimize(self, processed_batch: LLMBatch):
```

Comment:
I wont review the code since I have no idea about this part, but is `_and_maybe_optimize` part of the name still true?

Reply:
Yes, it optimizes only in training